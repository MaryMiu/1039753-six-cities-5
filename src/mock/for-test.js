export const offers = [{
  city: {
    name: `Cologne`,
    location: {
      latitude: 50.938361,
      longitude: 6.959974,
      zoom: 13
    }
  },
  images: [
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`
  ],
  title: `The house among olive `,
  rating: 3.8,
  type: `HOUSE`,
  bedrooms: 3,
  price: 493,
  goods: [
    `Laptop friendly workspace`,
    `Washer`,
    `Towels`,
    `Fridge`,
    `Air conditioning`,
    `Baby seat`,
    `Breakfast`
  ],
  host: {
    avatarUrl: `img/avatar-angelina.jpg`,
    isPro: true
  },
  description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
  location: {
    latitude: 50.957361,
    longitude: 6.9509739999999995,
    zoom: 16
  },
  id: 1,
  isPremium: false,
  isFavorite: false,
  maxAdults: 4,
  previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`
},
{
  city: {
    name: `Brussels`,
    location: {
      latitude: 50.846557,
      longitude: 4.351697,
      zoom: 13
    }
  },
  images: [
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/20.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`
  ],
  title: `Amazing and Extremely Central Flat`,
  rating: 4.3,
  type: `APARTMENT`,
  bedrooms: 3,
  price: 323,
  goods: [
    `Towels`,
    `Baby seat`,
    `Laptop friendly workspace`,
    `Dishwasher`,
    `Washing machine`,
    `Air conditioning`,
    `Washer`,
    `Coffee machine`,
    `Breakfast`,
    `Fridge`
  ],
  host: {
    avatarUrl: `img/avatar-angelina.jpg`,
    isPro: true
  },
  description: `This is a place for dreamers to reset, reflect, and create. Designed with a \`slow\` pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.`,
  location: {
    latitude: 50.839557,
    longitude: 4.346697,
    zoom: 16
  },
  id: 2,
  isPremium: false,
  isFavorite: false,
  maxAdults: 8,
  previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`
},
{
  city: {
    name: `Brussels`,
    location: {
      latitude: 50.846557,
      longitude: 4.351697,
      zoom: 13
    }
  },
  images: [
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`
  ],
  title: `The house among olive `,
  rating: 4.6,
  type: `APARTMENT`,
  bedrooms: 4,
  price: 218,
  goods: [
    `Washer`,
    `Laptop friendly workspace`,
    `Breakfast`,
    `Air conditioning`,
    `Baby seat`
  ],
  host: {
    avatarUrl: `img/avatar-angelina.jpg`,
    isPro: true
  },
  description: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
  location: {
    latitude: 50.865556999999995,
    longitude: 4.371696999999999,
    zoom: 16
  },
  id: 3,
  isPremium: false,
  isFavorite: false,
  maxAdults: 7,
  previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`
}
];

export const offer = {
  city: {
    name: `Cologne`,
    location: {
      latitude: 50.938361,
      longitude: 6.959974,
      zoom: 13
    }
  },
  images: [
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`
  ],
  title: `The house among olive `,
  rating: 3.8,
  type: `HOUSE`,
  bedrooms: 3,
  price: 493,
  goods: [
    `Laptop friendly workspace`,
    `Washer`,
    `Towels`,
    `Fridge`,
    `Air conditioning`,
    `Baby seat`,
    `Breakfast`
  ],
  host: {
    avatarUrl: `img/avatar-angelina.jpg`,
    isPro: true
  },
  description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
  location: {
    latitude: 50.957361,
    longitude: 6.9509739999999995,
    zoom: 16
  },
  id: 1,
  isPremium: false,
  isFavorite: false,
  maxAdults: 4,
  previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`
};

export const favoriteIds = [13, 15, 24, 26];

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const activeCity = `Paris`;

export const activeSortType = `Popular`;

export const review = {
  id: 1,
  user: {
    avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/6.jpg`,
    isPro: false
  },
  rating: 2,
  comment: `The room was spacious and clean. The pool looked nothing like the photos and desparately needs a clean. The sauna and spa were closed for lunar new year holiday.`,
  date: `2020-10-28T13:38:44.753Z`
};

export const reviews = [{
  id: 1,
  user: {
    avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/6.jpg`,
    isPro: false
  },
  rating: 2,
  comment: `The room was spacious and clean. The pool looked nothing like the photos and desparately needs a clean. The sauna and spa were closed for lunar new year holiday.`,
  date: `2020-10-28T13:38:44.753Z`
},
{
  id: 2,
  user: {
    avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/6.jpg`,
    isPro: false
  },
  rating: 3,
  comment: `What an amazing view! The house is stunning and in an amazing location. The large glass wall had an amazing view of the river!`,
  date: `2020-10-28T13:38:44.753Z`
},
{
  id: 3,
  user: {
    avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/2.jpg`,
    isPro: false
  },
  rating: 4,
  comment: `The deluxe room was a quite comfortable one with all the adequate facilities. The only thing that made me feel uncomfortable was the rude behavior of an impolite staff at the reception desk.`,
  date: `2020-10-28T13:38:44.753Z`
}
];

export const currentClasses = {
  listClass: `near-places__list`,
  cardClass: `near-places__card`,
  imgClass: `near-places__image-wrapper`
};

export const match = {
  path: `/offer/:id`,
  url: `/offer/11`,
  isExact: true,
  params: {
    id: `11`
  }
};

export const mapStyle = {
  display: `flex`,
  height: `100%`,
  width: 1144,
  margin: `0 auto`
};

export const offersId = [1, 2, 3];
