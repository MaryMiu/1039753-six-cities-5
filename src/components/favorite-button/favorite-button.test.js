import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import FavoriteButton, {FavoriteButton as FavoriteButtonWithoutStore} from "./favorite-button";
import configureStore from "redux-mock-store";
import {favoriteIds} from "../../mock/for-test";

const noop = () => {};

it(`FavoriteButton render correctly`, () => {
  const tree = renderer
    .create(<FavoriteButtonWithoutStore
      favoriteIds={favoriteIds}
      btnType={`place-card`}
      offerId={12}
      removeFavorite={noop}
      setFavorite={noop}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

describe(`Render connected to store component`, () => {
  const mockStore = configureStore([]);
  let store = null;
  let favoriteButtonComponent = null;

  beforeEach(() => {
    store = mockStore({
      FAVORITES: {
        favoriteIds,
      }
    });

    store.dispatch = jest.fn();

    favoriteButtonComponent = renderer.create(
        <Provider store={store}>
          <FavoriteButton
            btnType={`place-card`}
            offerId={12}
            removeFavorite={noop}
            setFavorite={noop}
          />
        </Provider>
    );
  });

  it(`Should FavoriteButton connected to store render correctly`, () => {
    expect(favoriteButtonComponent.toJSON()).toMatchSnapshot();
  });

  it(`Should call dispatch when button click`, () => {
    renderer.act(() => {
      favoriteButtonComponent.root.findByType(`button`).props.onClick();
    });

    expect(store.dispatch).toHaveBeenCalledTimes(1);
  });

});
