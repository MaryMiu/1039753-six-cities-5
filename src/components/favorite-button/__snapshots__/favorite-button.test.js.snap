// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteButton render correctly 1`] = `
<button
  className="place-card__bookmark-button button "
  onClick={[Function]}
  type="button"
>
  <svg
    className="place-card__bookmark-icon"
    height={19}
    width={18}
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    To bookmarks
  </span>
</button>
`;

exports[`Render connected to store component Should FavoriteButton connected to store render correctly 1`] = `
<button
  className="place-card__bookmark-button button "
  onClick={[Function]}
  type="button"
>
  <svg
    className="place-card__bookmark-icon"
    height={19}
    width={18}
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    To bookmarks
  </span>
</button>
`;
